internal/driver/fetch.go

```diff
@@ -107,6 +137,36 @@ func fetchProfiles(s *source, o *plugin.Options) (*profile.Profile, error) {
 	return p, nil
 }
 
+func grabSourceAndBases(sources, bases []profileSource, fetch plugin.Fetcher, obj plugin.ObjTool, ui plugin.UI) (*profile.Profile, *profile.Profile, plugin.MappingSources, plugin.MappingSources, bool, int, int, error) {
+	wg := sync.WaitGroup{}
+	wg.Add(2)
+	var p, pb *profile.Profile
+	var msrc, mbase plugin.MappingSources
+	var savesrc, savebase bool
+	var errsrc, errbase error
+	var countsrc, countbase int
+	go func() {
+		defer wg.Done()
+		p, msrc, savesrc, countsrc, errsrc = chunkedGrab(sources, fetch, obj, ui)
+	}()
+	go func() {
+		defer wg.Done()
+		pb, mbase, savebase, countbase, errbase = chunkedGrab(bases, fetch, obj, ui)
+	}()
+	wg.Wait()
+	save := savesrc || savebase
+
+	var err error
+	if errsrc != nil && errbase != nil {
```

Comment:
How about simplifying this: if there is a problem fetching source profiles - return that. Else if there is a problem returning base profiles - return that. Else return no error. I agree that printing out both errors when both errors happen may be nice, but I think it's very rarely required so it's better to keep the code simple.